faust/stores/aerospike.py

```diff
@@ -0,0 +1,198 @@
+import typing
+from typing import Any, Dict, Iterator, Optional, Tuple, Union
+
+try:  # pragma: no cover
+    import aerospike
+except ImportError:  # pragma: no cover
+    aerospike = None  # noqa
+
+from yarl import URL
+
+from faust.stores import base
+from faust.types import TP, AppT, CollectionT
+
+if typing.TYPE_CHECKING:  # pragma: no cover
+    from aerospike import SCAN_PRIORITY_MEDIUM, TTL_NEVER_EXPIRE, Client
+else:
+
+    class Client:  # noqa
+        """Dummy Client."""
+
+    TTL_NEVER_EXPIRE = -1
+    SCAN_PRIORITY_MEDIUM = 2
+
+
+if typing.TYPE_CHECKING:  # pragma: no cover
+    import aerospike.exception.RecordNotFound
+else:
+
+    class RecordNotFound:  # noqa
+        """Dummy Client."""
+
+
+aerospike_client: Client = None
+
+
+class AeroSpikeStore(base.SerializedStore):
+    """Aerospike table storage."""
+
+    client: Client
+    BIN_KEY = "value_key"
+    namespace: str = ""
+
+    def __init__(
+        self,
+        url: Union[str, URL],
+        app: AppT,
+        table: CollectionT,
+        options: typing.Mapping[str, Any] = None,
+        **kwargs: Any,
+    ) -> None:
+        try:
+            self.client = AeroSpikeStore.get_aerospike_client(options)
+            self.namespace = options.get("namespace", "")
+        except Exception as ex:
+            self.logger.error(f"Error configuring aerospike client {ex}")
+            raise ex
+        super().__init__(url, app, table, **kwargs)
+
+    @staticmethod
+    def get_aerospike_client(aerospike_config: Dict[Any, Any]) -> Client:
+        global aerospike_client
+        if aerospike_client:
+            return aerospike_client
+        else:
+            client = aerospike.client(aerospike_config)
+            try:
+                client.connect(None, None)
+                aerospike_client = client
+                return client
+            except Exception as e:
+                raise e
+
+    def _get(self, key: bytes) -> Optional[bytes]:
```

Comment:
Are _all_ keys always going to be `bytes`?
It seems like according to the Aerospike docs, we could make this a `str`, `int`, or `byte array`: https://www.aerospike.com/apidocs/python/client.html#aerospike-key-tuple

May be nice to make this flexible now, but no problem making it bytes now.

Reply:
this is a SerializedStore, so models are byte encoded/decoded in the `Table` framework